### PR TITLE
Beta.4

### LINKED ISSUE #677: Errors on Vanilla (Beta.2)
**Describe the bug**
There is a clear improvement on beta 2 compared to beta 1, however, we still have some errors: 
- see image (use switchNetwork + querySelector (UIHelpers.js))
- sometimes on Event "**NetworkCtrl.watch((network)...**" -> chain name = 'Chain1' instead of 'Ethereum' 

**FYI:**  We reused the component <w3-modal/> several times  (no reallocation)

So, another question: 
LOBSTR (Stellar) is compatible WalletConnect v2, but why is it not in your list of wallets?

**Envirionement Dev**
- Vue + Nuxt v2
- Use Web3Modal + Vanilla

**SDK Version**
 - Version [2.0.0-beta.2]

**To Reproduce**
```
  <template>
    <div>
      <w3m-modal />
    </div>
  </template>


 import {
  AccountCtrl,
  BalanceCtrl,
  ClientCtrl,
  ConfigCtrl,
  ModalCtrl,
  NetworkCtrl,
  TransactionCtrl
 } from '@web3modal/core'
 import { chains, providers } from '@web3modal/ethereum'
 import '@web3modal/ui'
  ...
  data: { 
    clientConfig: {
          projectId: PROJECT_ID,
          theme: 'dark',
          accentColor: 'green'
     }
    ethereumConfig: {
          appName: 'O21Pay',
          autoConnect: true,
          chains: [],
          providers: []
    },
   watchAccount: undefined,
  }
  
  async connect(chainId) {
      chainId = parseInt(chainId)

      if (chainId === 1) {
        this.ethereumConfig.chains = [chains.mainnet]
      } else if (chainId === 97) {
        this.ethereumConfig.chains = [chains.binanceSmartChainTestnet]
      } else {
        return
      }

      await ConfigCtrl.setConfig(this.clientConfig)
      await ClientCtrl.setEthereumClient(this.ethereumConfig)

      if (this.watchAccount) {
        this.watchAccount()
        this.watchAccount = undefined
      }
      // We force the network switch, because on the beta.1, "ClientCtrl.setEthereumClient(this.ethereumConfig)" was not 
      //   enough -> see chains
      NetworkCtrl.switchNetwork({ chainId })

      this.watchAccount = AccountCtrl.watch(async (account) => {
        console.log('******* Account ')
      })

      const account = AccountCtrl.get()
      if (account.isConnected === false) {
        ModalCtrl.open()
     }
  } 
```

**Expected behavior**
During the call of the method '**connect()**', we have systematically these 2 errors which are displayed...

**Screenshots**
(https://gist.github.com/Droppix/c6411d4c75ee42d89fc74f4fd85a6921)

**Desktop (please complete the following information):**
 - OS: [MacOS]
 - Browser [chrome]
 - Version [22]

Thanks

### PR DESCRIPTION
**Pull request by IljaDaderko**
⚠️ Don't merge until we coordinate with new docs release

- Removed custom modal hooks, we are now encouraging using wagmi directly
- Removed stateless hooks
- Removed hooks from react / react examples
- Renamed `useConnectModal` hook, to `useWeb3Modal`
- `ClientCtrl` now routes requests through generalised `.client()` that will be responsible for routing to ethereum / solana / cosmos etc... implementations
- `@web3modal/ethereum` package was drastically simplified, it now only exports functions required by modal ui package, everything else should be used from source i.e. wagmi or solib
- Added `encodeUriComponent` handling to link formatter helpers, so that it works with standalone mode
- Updated examples

closes #677 #661 #669 #682 #683 #697 #698 #699

### CODE REVIEW

File: chains/ethereum/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@web3modal/ethereum",
-  "version": "2.0.0-beta.3",
+  "version": "2.0.0-a38e60a",
```
Comment (devceline) [inline chains/ethereum/package.json]: Do we want to stick with this version?

File: chains/ethereum/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@web3modal/ethereum",
-  "version": "2.0.0-beta.3",
+  "version": "2.0.0-c2564bd",
```
Comment (devceline) [inline chains/ethereum/package.json:3]: Are we sure we want to stick with this version

File: examples/html/index.html [hunk 3]
```diff
@@ -2,22 +2,14 @@
 <html lang="en">
   <head>
     <meta charset="UTF-8" />
-    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <title>HTML Example</title>
-    <link rel="stylesheet" href="style.css" />
   </head>
```
Comment (devceline) [inline examples/html/index.html:6]: Why remove these

File: packages/ui/src/partials/w3m-core-button/index.ts [hunk 4]
```diff
@@ -8,14 +8,15 @@ import '../w3m-connect-button'
 @customElement('w3m-core-button')
 export class W3mCoreButton extends LitElement {
   // -- state & properties ------------------------------------------- //
-  @state() public isConnected = AccountCtrl.get().isConnected
+  @state() public isConnected = false
```
Comment (devceline) [inline packages/ui/src/partials/w3m-core-button/index.ts:11]: What are the implications of this change?

**Review by devceline — COMMENTED** [review]
Looks great, looked over every file.  It works. However, now the demo/example app does not have any sections to show off the app working. Maybe we can show Wagmi hooks working, or something along those lines because the example app works as implicit docs.

File: examples/html/index.html (hunk 3 above)
Comment (IljaDaderko) [reply examples/html/index.html:6]: All these removals were intentional to simplify the examples. I had same idea about wagmi section examples, but it is unnecessary chore for us to keep them updated as opposed to linking to wagmi. Instead, I am planning on full blown dapp example in our `web-examples` repo that showcases them at some point. But for examples within this repo, I think simplest possible ones do the trick.

File: packages/ui/src/partials/w3m-core-button/index.ts (hunk 4 above)
Comment (IljaDaderko) [reply packages/ui/src/partials/w3m-core-button/index.ts:11]: I moved this down into constructor (as it is now coming from client, there was ssr error for undefined ethereum client in next)

**Pull request merged**